### PR TITLE
feat: integrate solflare wallet

### PR DESCRIPTION
**Pull request by RyukTheCoder**
# Summary

Integrated Solflare wallet.

Fixes # 1468

# How did you test this change?

Tested by connecting and signing transaction on Solflare wallet extension.

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: wallets/provider-solflare/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+export const getSigners: (provider: any) => SignerFactory = signer;
+
+export const getWalletInfo: (allBlockChains: BlockchainMeta[]) => WalletInfo = (
+  allBlockChains
+) => {
+  const solana = solanaBlockchain(allBlockChains);
+  return {
+    name: 'Solflare',
+    img: 'https://raw.githubusercontent.com/rango-exchange/assets/main/wallets/phantom/icon.svg',
```
Comment (RanGojo) [inline wallets/provider-solflare/src/index.ts:51]: Please update the icon.

File: wallets/provider-solflare/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+const WALLET = WalletTypes.SOLFLARE;
+
+export const config = {
+  type: WALLET,
+};
+
+const walletInstance = new Solflare();
+
+export const getInstance = () => walletInstance;
```
Comment (RanGojo) [inline wallets/provider-solflare/src/index.ts:22]: When Solflare extension is not installed, your code doesn't detect it. I think it's better to have same behavior like other wallets.

File: wallets/provider-solflare/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  if (instance.publicKey) {
+    const account = instance.publicKey?.toString();
+
+    return {
+      accounts: [account],
+      chainId: Networks.SOLANA,
+    };
+  }
+  throw new Error('error');
```
Comment (RanGojo) [inline wallets/provider-solflare/src/index.ts:38]: Please use meaningful errors.

File: wallets/provider-solflare/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import { solanaBlockchain } from 'rango-types';
+
+import signer from './signer';
+
+const WALLET = WalletTypes.SOLFLARE;
+
+export const config = {
+  type: WALLET,
+};
+
```
Comment (RanGojo) [inline wallets/provider-solflare/src/index.ts:19]: It seems you didn't implement change account.

File: wallets/provider-solflare/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+) => {
+  const solana = solanaBlockchain(allBlockChains);
+  return {
+    name: 'Solflare',
+    img: 'https://raw.githubusercontent.com/rango-exchange/assets/main/wallets/phantom/icon.svg',
+    installLink: {
+      CHROME:
+        'https://chromewebstore.google.com/detail/solflare-wallet/bhhhlbepdkbapadjdnnojkbgioiodbic',
+
+      DEFAULT: 'https://solflare.com/',
```
Comment (RanGojo) [inline wallets/provider-solflare/src/index.ts:56]: Please add link for other browsers too.

**Review by RanGojo — CHANGES_REQUESTED** [review]

File: wallets/provider-solflare/src/index.ts (hunk 2 above)
Comment (RyukTheCoder) [reply wallets/provider-solflare/src/index.ts:22]: Tnx, handled it by checking the window for `solflare` object

File: wallets/provider-solflare/src/index.ts (hunk 3 above)
Comment (RyukTheCoder) [reply wallets/provider-solflare/src/index.ts:38]: Tnx, Fixed.

File: wallets/provider-solflare/src/index.ts (hunk 5 above)
Comment (RyukTheCoder) [reply wallets/provider-solflare/src/index.ts:56]: Done

File: wallets/provider-solflare/src/index.ts (hunk 4 above)
Comment (RyukTheCoder) [reply wallets/provider-solflare/src/index.ts:20]: Implemented it but it just works when switching to an account which you already have given the permission to connect to the app and you checked the "auto connect" option.

File: wallets/provider-solflare/src/index.ts (hunk 1 above)
Comment (RyukTheCoder) [reply wallets/provider-solflare/src/index.ts]: Fixed

**Review by RanGojo — APPROVED** [review]

**Pull request merged**